### PR TITLE
Add ut for flow_graph_dd_node to 100%

### PR DESCRIPTION
**Pull request by XuanYang-cn**
This PR:
- Change mutex to sync.Map
- Add ut for flow_graph_dd_node.go to 100%

See also: #6357

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

### CODE REVIEW

File: internal/datanode/flow_graph_dd_node.go [hunk 1]
```diff
@@ -102,12 +96,6 @@ func (ddn *ddNode) Operate(in []flowgraph.Msg) []flowgraph.Msg {
 					continue
 				}
 			}
```
Comment (sunby) [inline internal/datanode/flow_graph_dd_node.go:98]: can we change to 

``` golang

if imsg.CollectionID != ddn.collectionID || (msg.EndTs() < FilterThreshold && ddn.filterFlushedSegmentInsertMessages(imsg) ) {
     continue
}

```

File: internal/datanode/flow_graph_dd_node.go [hunk 2]
```diff
@@ -93,6 +81,12 @@ func (ddn *ddNode) Operate(in []flowgraph.Msg) []flowgraph.Msg {
 		case commonpb.MsgType_Insert:
 			log.Debug("DDNode with insert messages")
 			imsg := msg.(*msgstream.InsertMsg)
+			if imsg.CollectionID != ddn.collectionID {
+				//log.Debug("filter invalid InsertMsg, collection mis-match",
+				//	zap.Int64("Get msg collID", imsg.CollectionID),
+				//	zap.Int64("Expected collID", ddn.collectionID))
+				continue
+			}
```
Comment (sunby) [inline internal/datanode/flow_graph_dd_node.go:89]: can we change to

``` golang
if imsg.CollectionID != ddn.collectionID || (msg.EndTs() < FilterThreshold && ddn.filterFlushedSegmentInsertMessages(imsg) ) {
     continue
}
```

File: internal/datanode/flow_graph_dd_node_test.go [hunk 3]
```diff
@@ -12,19 +12,231 @@
...
+				test.inCollID,
+				&datapb.VchannelInfo{
+					FlushedSegments:   test.inFlushedSegs,
+					UnflushedSegments: []*datapb.SegmentInfo{di},
+				},
+			)
+
+			assert.Equal(t, "ddNode", ddNode.Name())
+			assert.Equal(t, test.inCollID, ddNode.collectionID)
+			assert.Equal(t, len(test.inFlushedSegs), len(ddNode.flushedSegments))
```
Comment (sunby) [inline internal/datanode/flow_graph_dd_node_test.go:61]: We can add a description field to test case and put these into a t.Run(), so we can find which failed case quickly.

File: internal/datanode/flow_graph_dd_node_test.go [hunk 4]
```diff
@@ -12,19 +12,231 @@
...
+		}{
+			{[]Msg{}},
+			{[]Msg{&insertMsg{}, &insertMsg{}, &insertMsg{}}},
+			{[]Msg{&insertMsg{}}},
+		}
+
+		for _, test := range invalidInTests {
+			ddn := ddNode{}
+			rt := ddn.Operate(test.in)
+			assert.Empty(t, rt)
```
Comment (sunby) [inline internal/datanode/flow_graph_dd_node_test.go:84]: same as above

File: internal/datanode/flow_graph_dd_node_test.go [hunk 5]
```diff
@@ -12,19 +12,231 @@
...
+		}
+
+		for _, test := range tests {
+			ddn := ddNode{
+				clearSignal:  test.ddnClearSignal,
+				collectionID: test.ddnCollID,
+			}
+
+			var createCollMsg msgstream.TsMsg = &msgstream.DropCollectionMsg{
+				DropCollectionRequest: internalpb.DropCollectionRequest{
```
Comment (sunby) [inline internal/datanode/flow_graph_dd_node_test.go:107]: same as above

**Review by XuanYang-cn — COMMENTED** [review]
Done

**Pull request merged**